packages/meditrak-app/app/entityMenu/actions.js

```diff
@@ -60,9 +61,30 @@ export const loadEntitiesFromDatabase = (isPrimaryEntity, questionId) => (
   const checkMatchesAttributeFilters = getEntityAttributeFilters(state, questionId);
   const filteredEntities = entities.filter(checkMatchesAttributeFilters);
 
+  const { questions, primaryEntityQuestionId, assessorId } = state.assessment;
+  const entityTypes = questions[primaryEntityQuestionId].config.entity.type;
+  const recentEntityIds = getRecentEntityIds(
+    database,
+    assessorId,
+    entityTypes,
+    state.country.selectedCountryId,
+  );
+  const recentEntityIdToIndex = {};
+  recentEntityIds.forEach((id, index) => {
+    recentEntityIdToIndex[id] = index;
+  });
+  const recentEntities =
+    recentEntityIds.length === 0
+      ? []
+      : database
+          .getEntities({ ...filters, id: recentEntityIds })
+          .slice()
+          .sort((a, b) => recentEntityIdToIndex[a.id] - recentEntityIdToIndex[b.id]);
```

Comment:
I found this bit quite confusing to get my head around. I think I was confused because recentEntityIdToIndex is created but it's not always used. Perhaps it could be done in a slightly more procedural way like this? But I think it's a pretty subjective preference.

```suggestion
  let recentEntities = [];

  if (recentEntityIds.length > 0) {
    const recentEntityIdToIndex = {};
    recentEntityIds.forEach((id, index) => {
      recentEntityIdToIndex[id] = index;
    });

    recentEntities = database
      .getEntities({ ...filters, id: recentEntityIds })
      .slice()
      .sort((a, b) => recentEntityIdToIndex[a.id] - recentEntityIdToIndex[b.id]);
  }
```

Reply:
Good call. I've gone with a slightly different option that I think makes it more readable still. No point making code denser than it needs, so thanks for the push back!